sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,200 @@ def transform(self, X):
+    array([[ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 0.,  1.,  3.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.],
+           [ 1.,  4.,  0.]])
+
+    See also
+    --------
+    https://blogs.technet.microsoft.com/machinelearning/2015/02/17/big-learning-made-easy-with-counts/      # noqa
+    """
+
+    def __init__(self, inclusion='all'):
+        self.inclusion = inclusion
+
+    @staticmethod
+    def _check_params(inclusion=None):
+        if (inclusion is not None and inclusion != 'all' and
+                not CountFeaturizer._valid_data_type(inclusion)):
+            raise ValueError("Illegal data type in inclusion")
+
+    @staticmethod
+    def _valid_data_type(type_check):
+        return type(type_check) == np.ndarray or type(type_check) == list \
+            or type(type_check) == set
+
+    def fit(self, X, y=None):
+        """Fits the CountFeaturizer to X, y
+
+        Stores the counts for each example X, conditional on y
+        Both X and y must be appropriately reshaped to a 2D list
+
+        Parameters
+        ----------
+        X : array
+            The data set to learn the counts from, conditional to 'y'
+
+        y : None, array
+            The 'y' that the counts are conditional to
+            - None (default) : The counts learned are not conditional on 'y'
+            - array : The counts learned are conditional on the given 'y'
+
+        Returns
+        -------
+        self
+        """
+
+        CountFeaturizer._check_params(inclusion=self.inclusion)
+
+        if y is not None:
+            X, y = check_X_y(X, y)
+            self.col_num_Y_ = len(y.take([0]))
+        else:
+            X = check_array(X)
+            self.col_num_Y_ = 0
+
+        len_data = len(X)
+
+        self.count_cache_ = defaultdict(lambda: defaultdict(int))
+        self.col_num_X_ = len(X[0])
+        self.y_set_ = set()
+
+        if type(self.inclusion) == str and self.inclusion == 'all':
+            inclusion_used = np.array(range(self.col_num_X_))
+        elif self.inclusion is None:
+            inclusion_used = np.array([])
+        elif CountFeaturizer._valid_data_type(self.inclusion):
+            inclusion_used = np.array(self.inclusion)
+        if y is not None:
+            for i in range(len_data):
+                X_key = tuple(X[i][j]
+                              for j in range(self.col_num_X_)
+                              if j in inclusion_used)
+                y_key = tuple(y.take([i]))
+                self.count_cache_[X_key][y_key] += 1
+                self.y_set_.add(y_key)
+        else:
+            self.y_set_.add(0)
+            for i in range(len_data):
+                X_key = tuple(X[i][j]
+                              for j in range(self.col_num_X_)
+                              if j in inclusion_used)
+                self.count_cache_[X_key][0] += 1
```

Comment:
How should this be implemented?
I had considered that notion, but I felt like this was better than initializing y as a list of [0, ..., 0] since it would take unnecessary memory, and it was better  than having an if-statement in a somewhat "critical" loop, which I think may be slow.